Sources/SourceControl/GitRepository.swift

```diff
@@ -387,6 +397,35 @@ public class GitRepository: Repository, WorkingCheckout {
         return localFileSystem.isDirectory(AbsolutePath(firstLine))
     }
 
+    /// Returns true if the file at `path` is ignored by `git`
+    public func areIgnored(_ paths: [AbsolutePath]) throws -> [Bool] {
+        return try queue.sync {
+            let stringPaths = paths.map({ $0.asString })
+
+            let pathsFile = try TemporaryFile()
+            try localFileSystem.writeFileContents(pathsFile.path) {
+                for path in paths {
+                    $0 <<< path.asString <<< "\0"
+                }
+            }
+
+            let args = [Git.tool, "-C", "\"\(self.path.asString)\"", "check-ignore", "-z", "--stdin", "<", "\"\(pathsFile.path.asString)\""]
```

Comment:
This might not be enough. We can use this method to properly escape: https://github.com/apple/swift-package-manager/blob/master/Sources/Basic/StringConversions.swift#L45